ngraph/src/ngraph/op/proposal.cpp

```diff
@@ -121,3 +121,78 @@ bool AttributeAdapter<op::ProposalAttrs>::visit_attributes(AttributeVisitor& vis
     return true;
 }
+
+constexpr NodeTypeInfo op::v4::Proposal::type_info;
+
+op::v4::Proposal::Proposal(const Output<Node>& class_probs,
+                           const Output<Node>& class_bbox_deltas,
+                           const Output<Node>& image_shape,
+                           const op::ProposalAttrs& attrs)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v4::Proposal::validate_and_infer_types()
+{
+    set_input_is_relevant_to_shape(2);
+
+    const auto& class_probs_pshape = get_input_partial_shape(0);
+    const auto& class_bbox_deltas_pshape = get_input_partial_shape(1);
+    const auto& image_shape_pshape = get_input_partial_shape(2);
+    if (class_probs_pshape.is_static() && class_bbox_deltas_pshape.is_static() &&
+        image_shape_pshape.is_static())
+    {
+        const Shape class_probs_shape{class_probs_pshape.to_shape()};
+        const Shape class_bbox_deltas_shape{class_bbox_deltas_pshape.to_shape()};
+        const Shape image_shape_shape{image_shape_pshape.to_shape()};
+
+        NODE_VALIDATION_CHECK(
+            this,
+            class_probs_shape.size() == 4,
+            "Proposal layer shape class_probs input must have rank 4 (class_probs_shape: ",
+            class_probs_shape,
+            ").");
+
+        NODE_VALIDATION_CHECK(this,
+                              class_bbox_deltas_shape.size() == 4,
+                              "Proposal layer shape class_bbox_deltas_shape input must have rank 4 "
+                              "(class_bbox_deltas_shape: ",
+                              class_bbox_deltas_shape,
+                              ").");
+
+        NODE_VALIDATION_CHECK(
+            this,
+            image_shape_shape.size() == 1,
+            "Proposal layer image_shape input must have rank 1 (image_shape_shape: ",
+            image_shape_shape,
+            ").");
+
+        NODE_VALIDATION_CHECK(
+            this,
+            image_shape_shape[0] >= 3 && image_shape_shape[0] <= 4,
+            "Image_shape 1D tensor must have => 3 and <= 4 elements (image_shape_shape[0]",
+            image_shape_shape[0],
+            ").");
+
+        auto batch_size = class_probs_shape[0];
+        set_output_type(0, get_input_element_type(0), Shape{batch_size * m_attrs.post_nms_topn, 5});
+        set_output_type(1, get_input_element_type(0), Shape{batch_size * m_attrs.post_nms_topn});
+    }
+    else
+    {
+        set_output_type(0, get_input_element_type(0), PartialShape::dynamic());
+        set_output_type(1, get_input_element_type(0), PartialShape::dynamic());
+    }
+}
+
+bool op::v4::Proposal::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("attrs", m_attrs);
+    return true;
+}
+
+std::shared_ptr<Node> op::v4::Proposal::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<Proposal>(new_args.at(0), new_args.at(1), new_args.at(2), m_attrs);
```

Comment:
Please, specify `op::v4::Proposal` instead of `Proposal`

Reply:
This was fiasco 🤦 Updated.